### PR TITLE
Support `void` and `Void` class lookups in `ReflectionUtils`

### LINKED ISSUE #4048: Support `void` class lookups in `ReflectionUtils`
Running following parametrized test in JUnit 5.11
```java
@ValueSource(strings = {
    "void", "boolean", "byte", "char", "short", "int", "long", "float", "double" })
@ParameterizedTest
void isPrimitive(Class<?> candidate) {
    assertTrue(candidate.isPrimitive());
}
```
only makes the "void" case fail.
```
org.junit.jupiter.api.extension.ParameterResolutionException:
    Error converting parameter at index 0: Failed to convert String "void" to type java.lang.Class
```
It looks like conversion of "void" to Class isn't supported.
Both `classNameToTypeMap` and `primitiveToWrapperMap` of `org.junit.platform.commons.util.ReflectionUtils` are lacking `void.class` and `Void.class`, but maybe this is by intention.

### PR DESCRIPTION
**Pull request by bjmi**
Fix #4048.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

**Comment by marcphilipp** [comment]
Could you please add an "integration test" demonstrating the scenario from #4048 is now supported?

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java [hunk 1]
```diff
@@ -223,6 +225,7 @@ public enum HierarchyTraversalMode {
 			Long[].class,
 			Float[].class,
 			Double[].class,
+			Void[].class,
```
Comment (sbrannen) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java:228]:
```suggestion
```

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java [hunk 2]
```diff
@@ -233,6 +236,7 @@ public enum HierarchyTraversalMode {
 			Long[][].class,
 			Float[][].class,
 			Double[][].class,
+			Void[][].class,
```
Comment (sbrannen) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java:239]:
```suggestion
```

**Review by sbrannen — CHANGES_REQUESTED** [review]
Thanks for the PR!

I would not consider `Void[]` and `Void[][]` "common types", so please remove those from the `classNameToTypeMap` cache.

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java (hunk 2 above)
Comment (bjmi) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java]: Applied.

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java (hunk 1 above)
Comment (bjmi) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java]: Applied.

**Review by marcphilipp — CHANGES_REQUESTED** [review]
> > Could you please add an "integration test" demonstrating the scenario from #4048 is now supported?
> 
> This is now implicitly tested by `DefaultArgumentConverterTests.convertsStringToClass()`. Should the newly introduced `DefaultArgumentConverterTests.IntegrationTests` still be retained?

Thanks! No, I don't think we need `DefaultArgumentConverterTests.IntegrationTests`.

Please add an entry to the 5.12.0-M1 release notes.

**Comment by bjmi** [comment]
> Thanks! No, I don't think we need `DefaultArgumentConverterTests.IntegrationTests`.

Removed.
> Please add an entry to the 5.12.0-M1 release notes.

As I'm not a native speaker feel free to change the text in the release notes.

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
Thanks, @bjmi! :+1:

**Pull request merged**